stripe/http_client.py

```diff
@@ -134,21 +134,33 @@ def __init__(
 
     # TODO: more specific types here would be helpful
     def request_with_retries(
-        self, method, url, headers, post_data=None
+        self,
+        method,
+        url,
+        headers,
+        post_data=None,
+        *,
+        _usage: Optional[List[str]] = None,
     ) -> Tuple[Any, int, Any]:
         return self._request_with_retries_internal(
-            method, url, headers, post_data, is_streaming=False
+            method, url, headers, post_data, is_streaming=False, _usage=_usage
         )
 
     def request_stream_with_retries(
-        self, method, url, headers, post_data=None
+        self,
+        method,
+        url,
+        headers,
+        post_data=None,
+        *,
+        _usage: Optional[List[str]] = None
     ) -> Tuple[Any, int, Any]:
         return self._request_with_retries_internal(
-            method, url, headers, post_data, is_streaming=True
+            method, url, headers, post_data, is_streaming=True, _usage=_usage
         )
 
     def _request_with_retries_internal(
-        self, method, url, headers, post_data, is_streaming
+        self, method, url, headers, post_data, is_streaming, *, _usage=None
```

Comment:
We're adding `_usage` as a keyword-only parameter in some places and positional in others - is there a reason for which places have which? Or should we unify on one?

Reply:
The only place it's positional-only I think is `_record_request_metrics` which is private and I think it should be required there.

Everywhere else's, it's a keyword argument and in some places I added the `*` but I kind of wanted to get the `*` thing figured out everywhere and I prepared https://github.com/stripe/stripe-python/pull/1144